src/common/exec_utils.h

```diff
@@ -621,6 +621,42 @@ inline nnvm::Graph AssignContext(nnvm::Graph g,
   return g;
 }
 
+/*!
+ * \brief infers shapes, dtypes, stypes, contexts for the forward graph
+ */
+inline nnvm::Graph InferForwardAttrs(nnvm::Graph g,
+                                     mxnet::ShapeVector arg_shapes,
+                                     nnvm::DTypeVector arg_dtypes,
+                                     StorageTypeVector arg_stypes,
+                                     const Context& default_ctx,
+                                     const std::map<std::string, Context>& ctx_map,
+                                     const std::vector<Context>& in_arg_ctxes,
+                                     const std::vector<Context>& aux_state_ctxes,
+                                     bool partial_shape = false) {
+  const auto& indexed_graph = g.indexed_graph();
+  const auto num_forward_inputs = indexed_graph.input_nodes().size();
+  g = AssignContext(g, default_ctx, ctx_map, in_arg_ctxes, {},
+                   aux_state_ctxes, {}, num_forward_inputs, g.outputs.size());
```

Comment:
this is the only consumer of default_ctx, is it really necessary to just do the shape/type prop that we're looking for pre-partitioning?

Reply:
AssignContext sets 'context' attribute to graph
https://github.com/apache/incubator-mxnet/blob/master/src/common/exec_utils.h#L512

which is used to set 'dev_mask' graph attribute later in InferStorageType
https://github.com/apache/incubator-mxnet/blob/master/src/executor/infer_graph_attr_pass.cc#L713-L719

Reply:
@mseth10 do we need to run InferStorageType? or can we skip it for the purpose of partitioning?

Reply:
As discussed offline, we need InferStorageType, which needs context.